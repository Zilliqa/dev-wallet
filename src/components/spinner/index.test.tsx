/**
 * This file is part of nucleus-wallet.
 * Copyright (c) 2018 - present Zilliqa Research Pte. Ltd.
 *
 * nucleus-wallet is free software: you can redistribute it and/or modify it under the
 * terms of the GNU General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later
 * version.
 *
 * nucleus-wallet is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * nucleus-wallet.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Spinner from '.';

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

test('matches the snapshot (small)', () => {
  const { container } = render(<Spinner size="small" data-testid="spinner" />);
  expect(container.firstChild).toMatchSnapshot();
});
test('matches the snapshot (medium)', () => {
  const { container } = render(<Spinner size="medium" data-testid="spinner" />);
  expect(container.firstChild).toMatchSnapshot();
});
test('matches the snapshot (large)', () => {
  const { container } = render(<Spinner size="large" data-testid="spinner" />);
  expect(container.firstChild).toMatchSnapshot();
});
