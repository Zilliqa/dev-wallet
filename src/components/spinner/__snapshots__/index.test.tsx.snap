// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot (large) 1`] = `
<div
  class="sc-bdVaJa gvsaAB"
  data-testid="spinner"
/>
`;

exports[`matches the snapshot (medium) 1`] = `
<div
  class="sc-bdVaJa dPYghA"
  data-testid="spinner"
/>
`;

exports[`matches the snapshot (small) 1`] = `
<div
  class="sc-bdVaJa kOGDNG"
  data-testid="spinner"
/>
`;
